### ISSUE TITLE
Awilix TypeError: Cannot read properties of undefined (reading 'cache')

### ISSUE THREAD

**Issue by krazyeu** [jeffijoe/awilix#345]
Hello, I am still new to Awilix infrastructure while developing together with MedusaJS. But Recently on both of my laptop (Mac and Window) after 2-3 days of development both hit this TypeError: Cannot Read properties of undefined (reading cache) as I can't continue working on the project's backend due to this issue. 

Please be chill with the criticism as this is my first-time opening issue ticket. 
The error log is below. 
```
error:   Error starting server
TypeError: Cannot read properties of undefined (reading 'cache')
    at Object.resolve (/Users/vincenty/Documents/Green Hero/GreenHero/greenhero/node_modules/@medusajs/utils/node_modules/awilix/lib/container.js:266:44)
    at MedusaModule.bootstrap (/Users/vincenty/Documents/Green Hero/GreenHero/greenhero/node_modules/@medusajs/modules-sdk/dist/medusa-module.js:137:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/vincenty/Documents/Green Hero/GreenHero/greenhero/node_modules/@medusajs/modules-sdk/dist/medusa-app.js:60:25
    at async Promise.all (index 1)
    at async loadModules (/Users/vincenty/Documents/Green Hero/GreenHero/greenhero/node_modules/@medusajs/modules-sdk/dist/medusa-app.js:40:5)
    at async MedusaApp (/Users/vincenty/Documents/Green Hero/GreenHero/greenhero/node_modules/@medusajs/modules-sdk/dist/medusa-app.js:146:24)
```
The error pointed towards 
awilix/lib/container.js
```
case lifetime_1.Lifetime.SINGLETON:
                    // Singleton lifetime means cache at all times, regardless of scope.
                    cached = rootContainer.cache.get(name);
                    if (!cached) {
                        resolved = resolver.resolve(container);
                        rootContainer.cache.set(name, { resolver, value: resolved });
                    }
                    else {
                        resolved = cached.value;
                    }
                    break;
                case lifetime_1.Lifetime.SCOPED:
```

**Comment by jeffijoe** [comment]
I would suggest opening an issue on the medusa issue tracker and link this issue. I’ll keep it open for now in case this is actually an awilix issue.

**Comment by vincent-yeo** [comment]
Issue solved, awilix package roll back to V8.0.1 works. 

[https://github.com/medusajs/medusa/issues/5606#issuecomment-1814968027](url)
